department-of-veterans-affairs/next-build
# Lighthous testing next

# Description
Adds Lighthouse testing to the next-build repo

## Ticket
No official ticket, just checking in code

## Developer Task

```[tasklist]
```

## Testing Steps

1. Run yarn lighthouse
2. verify file output in lighthouse directory

## QA steps

1. clone next build repo
2. run yarn install
3. run `yarn lighthouse`
4. verify lighthouse files that have been created

## Screenshots
Before:
After:


# Reviewer

### Reviewing a PR

This section lists items that need to be checked or updated when making changes to this repository. 

## Standard Checks

```[tasklist]
```

## Merging an Approved Layout

When merging a layout, you must ensure that the content type has been turned on for `next-build` inside the `CMS`. This CMS flag must be turned on for editors to preview their work using the next build preview server.

Resource types (layouts) that have not been approved by design should NOT be pushed to production. Ensure that [slug.tsx](../src/pages/[[...slug]].tsx) does not include your resource type if it is not approved.

The status of layouts should be kept up to date inside [templates.md](./templates.md). This includes QA progress, development progress, etc. A link should be provided for where testing can occur.

## Merging a Non-Approved Layout

Your new resource type should not be included inside [slug.tsx](../src/pages/[[...slug]].tsx). Items added here will go into production once merged into the `main` branch. It is imperative that we do not push anything live that has not been approved.

Ensure that this layout has been added to the [templates.md](./templates.md) file with the current status of the work.
